static/app/components/dashboards/widgetQueryFields.tsx

```diff
@@ -172,6 +152,60 @@ function WidgetQueryFields({
     return isLegalYAxisType(option.value.meta.dataType);
   };
 
+  const hideAddYAxisButton =
+    (['world_map', 'big_number'].includes(displayType) && fields.length === 1) ||
+    (['line', 'area', 'stacked_area', 'bar'].includes(displayType) &&
+      fields.length === 3);
+
+  const canDelete = fields.length > 1;
+
+  if (displayType === 'table') {
+    switch (widgetType) {
+      case WidgetType.ISSUE:
+        return (
+          <Field
+            data-test-id="columns"
+            label={t('Columns')}
+            inline={false}
+            style={{padding: `${space(1)} 0`, ...(style ?? {})}}
+            error={errors?.fields}
+            flexibleControlStateSize
+            stacked
+            required
+          >
+            <StyledColumnEditCollection
+              columns={fields}
+              onChange={handleColumnChange}
+              fieldOptions={fieldOptions}
+              organization={organization}
+              source={widgetType}
+            />
+          </Field>
+        );
+      case WidgetType.DISCOVER:
+      default:
+        return (
+          <Field
+            data-test-id="columns"
+            label={t('Columns')}
+            inline={false}
+            style={{padding: `${space(1)} 0`, ...(style ?? {})}}
+            error={errors?.fields}
+            flexibleControlStateSize
+            stacked
+            required
+          >
+            <StyledColumnEditCollection
+              columns={fields}
+              onChange={handleColumnChange}
+              fieldOptions={fieldOptions}
+              organization={organization}
+            />
+          </Field>
+        );
+    }
+  }
```

Comment:
Can this be simplified to something like this?
```suggestion
  if (displayType === 'table') {
    return (
        <Field
            data-test-id="columns"
            label={t('Columns')}
            inline={false}
            style={{padding: `${space(1)} 0`, ...(style ?? {})}}
            error={errors?.fields}
            flexibleControlStateSize
            stacked
            required
        >
            <StyledColumnEditCollection
                columns={fields}
                onChange={handleColumnChange}
                fieldOptions={fieldOptions}
                organization={organization}
                source={widgetType === WidgetType.ISSUE ? widgetType : undefined}
            />
        </Field>
    );
  }
```